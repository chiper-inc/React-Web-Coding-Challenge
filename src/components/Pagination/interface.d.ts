export interface IPagination {
  currentPage: number;
  pages: number;
  onSubmit: (page: number) => {};
}
