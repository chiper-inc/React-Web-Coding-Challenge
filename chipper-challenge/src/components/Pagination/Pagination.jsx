import React from 'react';
import { useSelector } from 'react-redux';
import { StyledDiv } from './styled';
import PropTypes from 'prop-types';

const Pagination = ({ paginate, cardsPerPage }) => {
    const bikes = useSelector((state) => state.stolenBikes);

    let pageNumbers = [];

    for (let i = 1; i <= Math.ceil(bikes.length / cardsPerPage); i++) {
        pageNumbers.push(i);
    }

    return (
        <StyledDiv className='pagination'>
            <ul>
                {pageNumbers.map((number) => (
                    <li key={number}>
                        <a onClick={() => paginate(number)} href='#'>
                            {number}
                        </a>
                    </li>
                ))}
            </ul>
        </StyledDiv>
    );
};

Pagination.propTypes = {
    paginate: PropTypes.number,
    cardsPerPage: PropTypes.number,
};

export default Pagination;
