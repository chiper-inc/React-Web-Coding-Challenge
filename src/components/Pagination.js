import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
  display: inline-block;
  border-radius: 6px;
  padding: 0.5rem 0;
  margin: 0.5rem 0 0.5rem 1rem;
  width: 11rem;
  background: #00264a;
  color: white;
  border: 2px solid white;
  cursor: pointer;
`

export const Pagination = ({ handlerNext, handlerPrev }) => {

    return (
        <>
            <Button onClick={() => handlerPrev()}>Prev</Button>
            <Button onClick={() => handlerNext()}>Next</Button>

        </>
    )
}
