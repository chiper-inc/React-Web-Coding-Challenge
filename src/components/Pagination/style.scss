.pagination {
  @apply flex mt-8 flex-col md:flex-row;

  .btn {
    @apply px-12 py-3 m-4 ml-0 bg-blue-400 text-white font-bold rounded;
    &:hover {
      @apply bg-blue-600;
    }
    &.desactive {
      @apply cursor-not-allowed;
      pointer-events: none;
    }
  }

  .item-number {
    @apply p-4 m-4 ml-0 bg-gray-400 text-white font-bold rounded cursor-pointer hidden md:block;
    &:hover {
      @apply bg-blue-600;
    }
    &.active {
      @apply bg-blue-400;
    }
  }
}
