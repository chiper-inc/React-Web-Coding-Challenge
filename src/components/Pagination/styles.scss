.Pagination {
  width: 80%;
  height: 100px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 30px;
  
  button {
    margin-right: 10px;
  }

  span {
    margin: 10px;
    font-weight: 600;
  }
}
