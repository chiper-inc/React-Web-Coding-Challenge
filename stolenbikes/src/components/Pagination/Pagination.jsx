import React from "react";

function pagination({ casesPerPage, totalCasesCount,paginate }) {
  const pageNumbers = [];

  for (let i = 1; i <= Math.ceil(totalCasesCount / casesPerPage); i++) {
    pageNumbers.push(i);
  }
  return (
    <nav>
      <ul className="page-numbers">
        {pageNumbers.map((number) => (
          <li key={number}>
            <button onClick={()=> paginate(number)}>
              {number}
            </button>
          </li>
        ))}
      </ul>
    </nav>
  );
}

export default pagination;
