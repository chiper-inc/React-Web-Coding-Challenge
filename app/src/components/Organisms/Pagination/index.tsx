import { globalStyles } from '@/styles/globalStyles';
import React from 'react';
import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
import { Nav, List, Item, Previous, Next } from './styles';

export interface ItemProps {
  disabled?: boolean;
  active?: boolean;
  onClick?: (event: any) => void;
}

interface PaginationProps {
  color?: string;
  current: number;
  total: number;
  handleClick: (num: number) => void;
}

const Pagination: React.FC<PaginationProps> = ({
  current,
  total,
  color = globalStyles.colors.primary,
  handleClick,
}: PaginationProps) => {
  const [currentArray, setCurrentArray] = React.useState<Array<number>>([]);

  const handleClickPage = (page: number): void => {
    page >= 1 && page <= total
      ? handleClick(page)
      : page < 1
      ? handleClick(1)
      : handleClick(total);
  };

  React.useEffect(() => {
    const init =
      current === total && current > 1
        ? current - 4
        : current >= total - 4 && current !== 1
        ? current - 4
        : current > 3
        ? current - 2
        : 1;
    const arrContainer = [];
    for (let i = init; i < init + 5; i++) {
      arrContainer.push(i);
    }
    setCurrentArray(arrContainer);
  }, [current, total]);

  return (
    <Nav>
      <List>
        <Item
          color={color}
          disabled={current === 1}
          onClick={() => current !== 1 && handleClickPage(current - 1)}
        >
          <Previous>
            <FaChevronLeft />
          </Previous>
        </Item>
        {total > 5 && current > 3 && (
          <>
            <Item
              active={current === 1}
              color={color}
              onClick={() => handleClickPage(1)}
            >
              <span>1</span>
            </Item>
            <span>...</span>
          </>
        )}
        {currentArray.map((item, index) => (
          <Item
            key={index}
            active={current === item}
            color={color}
            onClick={() => handleClickPage(item)}
          >
            <span>{item}</span>
          </Item>
        ))}
        {total > 5 && current < total && (
          <>
            <span>...</span>
            <Item
              active={current === total}
              color={color}
              onClick={() => handleClickPage(total)}
            >
              <span>{total}</span>
            </Item>
          </>
        )}
        <Item
          color={color}
          disabled={current === total}
          onClick={() => current !== total && handleClickPage(current + 1)}
        >
          <Next>
            <FaChevronRight />
          </Next>
        </Item>
      </List>
    </Nav>
  );
};

export { Pagination };
