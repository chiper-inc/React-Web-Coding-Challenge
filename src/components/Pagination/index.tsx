import { useBikeIndex } from "../../network/api/hooks/useBikeIndex";
import { Button } from "../StyledComponents/Button";
import { IPagination } from "./interface";
import './styles.scss';

const Pagination = ({ currentPage, pages, onSubmit }: IPagination) => {
  return (
    <section className="Pagination">
    <Button onClick={()=> onSubmit(1)}>{`<< First`}</Button>
    <Button onClick={()=> onSubmit(currentPage > 1 ? currentPage - 1 : currentPage)}>{`< Prev`}</Button>
    <span> {currentPage} / {pages}</span>
    <Button onClick={()=> onSubmit(currentPage < pages ? currentPage + 1 : currentPage)}>{`> Next`}</Button>
    <Button onClick={()=> onSubmit(pages)}>{`>> Last`}</Button>
    </section>
  )
}

export default Pagination;
