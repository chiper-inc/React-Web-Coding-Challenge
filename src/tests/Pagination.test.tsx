import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../components/Pagination';

test('renders the component Pagination Test', () => {
  // const [page, setPage] = React.useState('');
  const setPage = jest.fn();
  const component = shallow(
    <Pagination page={1} setPage={() => setPage()} totalPage={7} />
  );
  expect(component).toMatchSnapshot();
});
