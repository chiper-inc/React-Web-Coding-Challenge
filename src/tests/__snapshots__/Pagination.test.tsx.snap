// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component Pagination Test 1`] = `
<div
  className="pagination flex"
>
  <button
    className="btn"
    onClick={[Function]}
  >
    First
  </button>
  <button
    className="btn desactive"
    disabled={true}
    onClick={[Function]}
  >
    Prev
  </button>
  <button
    className="btn"
    disabled={false}
    onClick={[Function]}
  >
    Next
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    Last
  </button>
</div>
`;
